apps/web/ee/components/apiKeys/ApiKeyDialogForm.tsx

```diff
@@ -0,0 +1,165 @@
+  const [neverExpires, setNeverExpires] = useState(false);
+  const handleNoteChange = (e: { target: { value: string | null } }) => {
+    form.setValue("note", e.target.value);
+  };
+  const {
+    defaultValues = {
+      note: "" as string | undefined,
+      expiresAt: dayjs().add(1, "month").toDate(),
+    } as Omit<TApiKeys, "userId" | "createdAt" | "lastUsedAt">,
+  } = props;
+
+  const [selectedDate, setSelectedDate] = useState(dayjs().add(1, "month").toDate());
+  const [newApiKey, setNewApiKey] = useState("");
+  const [successfulNewApiKeyModal, setSuccessfulNewApiKeyModal] = useState(false);
+  const [newApiKeyDetails, setNewApiKeyDetails] = useState({
+    id: "",
+    hashedKey: "",
+    expiresAt: null as Date | null,
+    note: "" as string | null,
+  });
+
+  const handleDateChange = (e: SetStateAction<Date> | null) => {
+    setSelectedDate(e);
+    form.setValue("expiresAt", e);
+  };
+
+  const form = useForm({
+    defaultValues,
+  });
+  return (
+    <>
+      {successfulNewApiKeyModal ? (
+        <>
+          <div className="mb-10">
+            <h2 className="font-semi-bold font-cal mb-2 text-xl tracking-wide text-gray-900">
+              {" "}
+              {t("success_api_key_created")}
+            </h2>
+            <div className="text-sm text-gray-900">
+              <span className="font-semibold">{t("success_api_key_created_bold_tagline")}</span>{" "}
+              {t("you_will_not_view_once_close_modal")}
+            </div>
+          </div>
+          <div>
+            <div className="flex">
+              <code className="my-2 mr-1 w-full truncate rounded-sm bg-gray-100 py-2 px-3 align-middle font-mono text-gray-800">
+                {newApiKey}
+              </code>
+              <Tooltip content={t("copy_to_clipboard")}>
+                <Button
+                  onClick={() => {
+                    navigator.clipboard.writeText(newApiKey);
+                    showToast(t("api_key_copied"), "success");
+                  }}
+                  type="button"
+                  className=" my-2 px-4 text-base">
+                  <ClipboardCopyIcon className="mr-2 h-5 w-5 text-neutral-100" />
+                  {t("copy")}
+                </Button>
+              </Tooltip>
+            </div>
+            {/* // FIXME: Don't hardcode en-GB locale for date formatting */}
+            <span className="text-sm text-gray-400">
+              {" "}
+              Expires {newApiKeyDetails?.expiresAt?.toLocaleDateString("en-GB", options)}
+            </span>
+          </div>
+          <DialogFooter>
+            <Button type="button" color="secondary" onClick={props.handleClose} tabIndex={-1}>
+              {t("done")}
+            </Button>
+          </DialogFooter>
+        </>
+      ) : (
+        <Form
+          form={form}
+          handleSubmit={async (event) => {
+            const newApiKey = await utils.client.mutation("viewer.apiKeys.create", event);
```

Comment:
We should use try catch on awaits

Reply:
I removed the .catch() from the call because I was just console.log(error), what should I do in the catch.

Reply:
Added and showToast with error, need to test if error passess message though.